### PR TITLE
Add CondIsSlimeChunk

### PR DESCRIPTION
**Pull request by Nicofisi**
Target Minecraft versions: tested on Paper 1.12.2 and Spigot 1.8.8
Requirements: none
Related issues: #678

### CODE REVIEW

File: src/main/java/ch/njol/skript/conditions/CondIsSlimeChunk.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+
+/**
+ * @author Nicofisi
+ */
+@Name("Is Slime Chunk")
+@Description({"Tests whether a chunk is a so-called slime chunk.",
+		"Slimes can generally spawn in the swamp biome and in slime chunks.",
+		"For more info, see https://minecraft.gamepedia.com/Slime#.22Slime_chunks.22"})
```
Comment (Nicofisi) [inline src/main/java/ch/njol/skript/conditions/CondIsSlimeChunk.java]: And in the end I forgot to make it `<a href=...>`.. this should be `<a href=...>`, right?

Comment (Snow-Pyon) [reply src/main/java/ch/njol/skript/conditions/CondIsSlimeChunk.java]: Yes, it should.

**Comment by TheLimeGlass** [comment]
`([a] slime chunk|slime chunks|slimey [chunk[s]])`

Also `chunk.isSlimeChunk()` is a thing, but wasn't added until 1.11.

Also there is a NMS method for all of this, but Skript doesn't use reflection

NMS Chunk source:
```
public Random a(long i) {
        return new Random(this.world.getSeed() + (long) (this.locX * this.locX * 4987142) + (long) (this.locX * 5947611) + (long) (this.locZ * this.locZ) * 4392871L + (long) (this.locZ * 389711) ^ i);
    }
```
EntitySlime source:
```
@Override
public boolean isSlimeChunk() {
    // 987234911L is deterimined in EntitySlime when seeing if a slime can spawn in a chunk
    return getHandle().a(987234911L).nextInt(10) == 0;
}
```

**Comment by TheLimeGlass** [comment]
Also Paper has an option to make all chunks slime chunks, so this will be invalid if that option is true.

**Comment by bensku** [comment]
I would recommend calling Bukkit to see if chunk is slimy. You can keep your code as fallback for 1.8 if you wish. Also, unless you literally reverse-engineered that algorithm yourself, mention the source in comment *and* make sure it allows licensing under GPLv3.

**Comment by Nicofisi** [comment]
The algorithm is from the link in the description, from the wiki.. I have no idea on what license the wiki is, I'd sort of just call it public knowledge /shrug

**Comment by Nicofisi** [comment]
> Also chunk.isSlimeChunk() is a thing, and was added in March 2016 which was 1.9, and that's what this Skript fork supports.

From [this comment](https://github.com/SkriptLang/Skript/issues/678#issuecomment-309714019) by @Syst3ms I assumed it's only 1.11+?

https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/319013857d3eb9a64171d64e453f36dd0641447f

> 	Senmori committed 319013857d3 - 23 Mar **2017**
Add Chunk.isSlimeChunk to check if a chunk can spawn slimes

**Comment by bensku** [comment]
Wiki is licensed under CC BY-NC-SA, which is not compatible with GPLv3. But if you think it can be considered public domain, I think we can merge it.

**Comment by Snow-Pyon** [comment]
@Nicofisi if you change it to call Chunk#isSlimeChunk and add a note on the documentation clarifying this:
`Paper has an option to make all chunks slime chunks, so this will be invalid if that option is true.`

It should be good enough to merge.

**Comment by Nicofisi** [comment]
Does the paper `isSlimeChunk` method take the biome already into consideration? Should the whole condition take the biomes into consideration? (swamp/mushroom island)

**Comment by Snow-Pyon** [comment]
@Nicofisi it does the exact same calculation the Minecraft server does to know whether a chunk is a slime chunk.

The reason why I personally don't want the harcoded calculation for it is:
1. The way slime chunks are defined could change with the time, something not very feasible but it could happen and if that's the case the condition would just not be useful at all. Let the Spigot guys handle that and use their API.
2. The fact that Wiki is licensed under a license that isn't compatible with GPL is kinda, you know, problematic. We're safe using the Spigot method.

**Comment by Nicofisi** [comment]
@Snow-Pyon alright but just confirming - that means no support for <1.11. Are you sure?

**Comment by Snow-Pyon** [comment]
@Nicofisi welp, you could left the code as fallback for the version that don't have it like bensku mentioned.

**Comment by TheBentoBox** [comment]
Leaving the proper formula as a fallback is fine. The main downside to using the hard coded formula is that it could change but that's irrelevant for old versions so I don't see a downside.

Also the formula from the wiki is taken directly from the MC code itself. I don't think they really have any license on it.

**Comment by Pikachu920** [comment]
well, MC's code is proprietary so using it is incompatible with GPL... the question is: is an equation really part of MC's code? Generally, I think the answer is no.

**Comment by TheLimeGlass** [comment]
It's open source, how is it against the GPL if Skript and Spigot are open source? Bensku was talking about the wiki it was taken from.

**Pull request merged**